src/sql/workbench/parts/notebook/models/notebookModel.ts

```diff
@@ -292,7 +297,24 @@ export class NotebookModel extends Disposable implements INotebookModel {
 				this._defaultLanguageInfo = this.getDefaultLanguageInfo(contents);
 				this._savedKernelInfo = this.getSavedKernelInfo(contents);
 				if (contents.cells && contents.cells.length > 0) {
-					this._cells = contents.cells.map(c => factory.createCell(c, { notebook: this, isTrusted: isTrusted }));
+					this._cells = contents.cells.map(c => {
+						let cellContent = (<nb.ICellContents>c);
+						if (c && cellContent && cellContent.cell_type === CellTypes.Markdown) {
+							this._textCellsLoading++;
+						}
+						let cellModel = factory.createCell(c, { notebook: this, isTrusted: isTrusted });
```

Comment:
Let's refactor this for easier reading:
- Move `let cellModel = factory.createCell(c, { notebook: this, isTrusted: isTrusted });` to be the 1st line, you always do this
- Move the telemetry tracking to its own method so it does't interfere / get confused with the basic model load flow